src/studio-home/collapsible-state-with-action/index.jsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import {
+  Collapsible,
+  Bubble,
+  Icon,
+  StatefulButton,
+} from '@edx/paragon';
+import {
+  Add as AddIcon,
+  Minus as MinusIcon,
+} from '@edx/paragon/icons/es5';
+import { useIntl } from '@edx/frontend-platform/i18n';
+
+import { RequestStatus } from '../../data/constants';
+import { COURSE_CREATOR_STATES, STATEFUL_BUTTON_STATES } from '../../constants';
+import { getStudioHomeData, getSavingStatuses } from '../data/selectors';
+import { requestCourseCreatorQuery } from '../data/thunks';
+import messages from './messages';
+
+const CollapsibleStateWithAction = ({ state, className }) => {
+  const intl = useIntl();
+  const dispatch = useDispatch();
+  const {
+    platformName,
+    studioName,
+    studioShortName,
+  } = useSelector(getStudioHomeData);
+  const { courseCreatorSavingStatus } = useSelector(getSavingStatuses);
+
+  const requestButtonStates = {
+    labels: {
+      default: intl.formatMessage(messages.unrequestedCollapsibleDefaultButton),
+      pending: intl.formatMessage(messages.unrequestedCollapsiblePendingButton),
+      error: intl.formatMessage(messages.unrequestedCollapsibleFailedButton),
+    },
+    disabledStates: [STATEFUL_BUTTON_STATES.pending, STATEFUL_BUTTON_STATES.error],
+  };
+
+  // eslint-disable-next-line no-nested-ternary
+  const requestButtonCurrentState = courseCreatorSavingStatus === RequestStatus.PENDING
+    ? STATEFUL_BUTTON_STATES.pending
+    : courseCreatorSavingStatus === RequestStatus.FAILED
+      ? STATEFUL_BUTTON_STATES.error
+      : STATEFUL_BUTTON_STATES.default;
```

Comment:
Nested ternary expressions are hard to read. I propose to create constants and simplify this code.

Reply:
changed